src/xref2/env.ml

```diff
@@ -23,13 +23,13 @@ type resolver = {
 let unique_id = ref 0
 
 type lookup_type =
-  | Module of Odoc_model.Paths_types.Identifier.reference_module * bool
-  | ModuleType of Odoc_model.Paths_types.Identifier.module_type * bool
+  | Module of Odoc_model.Paths_types.Identifier.reference_module
+  | ModuleType of Odoc_model.Paths_types.Identifier.module_type
   | RootModule of
       string
       * [ `Forward | `Resolved of Digest.t ] option
   | ModuleByName of
-      string * Odoc_model.Paths_types.Identifier.reference_module option
+      string * Odoc_model.Paths_types.Identifier.reference_module
```

Comment:
This PR removes the feature of checking lookups that failed. (they are no longer recorded)
Is this an issue?

Reply:
Good question! I'll have a little look.

Reply:
Hmm, I think it might make a difference. When we look up modules we check first if it's in the `modules` map, and only if it's not do we lookup root modules (which we do by name). I think to be on the safe side it would be better if we could keep recording the failures until we're completely satisfied with the model. At that point we could remove this check and see if it change the output at all (I'm assuming here that debugging broken memoizing is going to be a nightmare!)

Reply:
There is an issue with scopes like `label_parent` and `signature` which may match modules never look at root modules. I'll reimplement recording and fix this issue.